src/providers/info.rs

```diff
@@ -107,6 +109,33 @@ impl InfoProvider for DefaultInfoProvider<'_> {
             logo_url: format!("{}/{}", url, manifest.icon_path),
         })
     }
+
+    fn address_info(&mut self, address: &str) -> Result<AddressInfo> {
+        self.token_info(address)
+            .map(|it| AddressInfo {
+                name: it.name,
+                logo_uri: it.logo_uri.to_owned(),
+            })
+            .or_else(|_| {
+                let url = format!(
+                    "{}/v1/contracts/{}/",
+                    base_transaction_service_url(),
+                    address
+                );
+                let contract_info_json =
+                    self.cache
+                        .request_cached(self.client, &url, safe_app_manifest_cache())?;
```

Comment:
not sure if we should use this cache timeout here, very different use cases 🤔

Reply:
It was just the only available super long cache duration, I was thinking we could clean this up with the task you mentioned regarding having cache durations named after their duration rather than the functionality in which they are used.